PLNPlugin.inc.php

```diff
@@ -637,4 +638,22 @@ protected function _sendFile($method, $url, $filename) {
 			'result' => (string) $response->getBody(),
 		);
 	}
+
+	/**
+	 * Disables the session whenever the request is supposed to be handled by the plugin
+	 * This is required to avoid a disabled journal from being redirected at the PKPPageRouter::route()
+	 * Note: Adjusting the context temporarily through $context->set('enabled', true) doesn't work here,
+	 * because its reference is reloaded later on, when we have no other way to intercept it
+	 */
+	private function _disableSession(): void
+	{
+		$request = $this->getRequest();
+		$page = $request->getRequestedPage();
+		$operation = $request->getRequestedOp();
+		$arguments = $request->getRequestedArgs();
+
+		if ([$page, $operation] === ['pln', 'deposits'] || [$page, $operation, reset($arguments)] === ['gateway', 'plugin', 'PLNGatewayPlugin']) {
+			SessionManager::disable();
```

Comment:
Here's the redirection for disabled journals: https://github.com/pkp/pkp-lib/blob/main/classes/core/PKPPageRouter.inc.php#L229-L237